src/hotkey/command_executor.cpp

```diff
@@ -516,10 +516,13 @@ static void event_execute( const SDL_Event& event, command_executor* executor)
 		return;
 	}
 
+	const hotkey_command& command = get_hotkey_command(hk->get_command());
 	bool press = event.type == SDL_KEYDOWN ||
 			event.type == SDL_JOYBUTTONDOWN ||
-			event.type == SDL_MOUSEBUTTONDOWN ||
-			event.type == SDL_TEXTINPUT;
+			event.type == SDL_MOUSEBUTTONDOWN;
+	if(!command.toggle) {
+		press = press || event.type == SDL_TEXTINPUT;
+	}
 
 	execute_command(hotkey::get_hotkey_command(hk->get_command()), executor, -1, press);
```

Comment:
Please eliminate the now-duplicate call to `hotkey::get_hotkey_command()`:

    execute_command(command, executor, -1, press);